### DISCUSSION TITLE
[Idea] Sandbox backoffice SQL Manager behind a Feature Flag

### DISCUSSION THREAD

**Discussion by matks** [PrestaShop/PrestaShop#32711] [Ideas]
### The problem

The backoffice SQL Manager page can be very useful as it allow backoffice users to write and store SQL queries that can be run at will. It can help for exploring data and exporting data.

However the idea behind it is not safe. The SQL Manager allows backoffice users to run SQL queries in read-only. No data modification is allowed by this tool.

Multiple restrictions have been implemented to make sure this is indeed read-only. But this is just code.

After working more than 10 years in web software development I know no amount of code can protect you against smart people finding ways to bypass these restrictions and run queries that do modify the database content. Here's a recent one https://github.com/PrestaShop/PrestaShop/security/advisories/GHSA-p379-cxqh-q822 and I know there will be more.

In the great cat-and-mouse game of software security, nothing is 100% safe ever.

### The solution

We should allow backoffice users to disable the SQL manager, making it un-accessible and consequently removing any risk. A simple switch in the backoffice (I think a feature flag can do the job) would allow people who consider it very useful (and accept the risk that comes with the usage) to enable it again.

To sum up, for new shops
- SQL Manager should be disabled by default
- A switch in the backoffice allows to enable it

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
This feature flag should be available to change only via the environment variable. If you have access to the back office, most of the time, it is a profile with `SuperAdmin` rights... really not that many shops use permissions in PrestaShop nor know how to configure them. 

We have more and more reports for back office vulnerabilities, but let's be honest, if attackers gain access to the back office, and - very often if they do, it's a `SuperAdmin`, they can do whatever they want simply by installing a new module.

Maybe we should move directly to https://github.com/PrestaShop/PrestaShop/discussions/32712 idea.

**Reply by matthieu-rolland** [reply]
I agree with you about moving directly to #32712
